File: prepare/leetcode/src/lc/s1/p00/Runner.py
from typing import List
import fileinput

def main():
    sol = Solution()
    with open('in.txt') as f:
        t = int(f.readline().strip())
        for i in range(1,t+1):
            n = int(f.readline().strip())
            arr = [list(map(int, f.readline().strip().split() )) for i in range(n)]
            print ("Case #{}: {}".format(i, sol.solve(arr)))

class Solution:
    def solve(self, arr):
        n = len(arr)
        trace = sum([arr[i][i] for i in range(n)])
        rows = sum([1 if len(set([arr[i][j] for j in range(n)])) != n else 0 for i in range(n)])
        cols = sum([1 if len(set([arr[j][i] for j in range(n)])) != n else 0 for i in range(n)])
        return ' '.join([str(trace), str(rows), str(cols)])

if __name__ == "__main__":
    main()

File: prepare/leetcode/src/lc/s1/p00/Runner.java
package lc.s1.p00;


import util.IO;

public class Runner {
	
	public static void main(String[] args) {
		IO.scan(Solution.class, "in.txt", in -> {
			int t = in.nextInt();
			for (int i = 1; i <= t; ++i) {
				int n = in.nextInt();
				int[][] matrix = new int[n][n];
				for (int r = 0; r < n; r++) {
					for (int c = 0; c < n; c++) {
						matrix[r][c] = in.nextInt();
					}
				}
				System.out.println("Case #" + i + ": " + Solution.solve(n, matrix));
			}
		});
	}

}
